### DISCUSSION TITLE
Updating WindowsAppSDK in Uno.WinUI sln gives build errors CS0433 type '...' exists in both Microsoft.WinUI and Uno.UI ?

### DISCUSSION THREAD

**Discussion by VincentH-Net** [unoplatform/uno#13569] [Q&A] [answered]
When I create a new Uno Platform App in VS with the latest Uno Platform VS extension (4.10.10.11), Blank startup type accepting defaults, all builds fine.

However, when I update `Microsoft.WindowsAppSDK` from `1.3.230724000` to `1.4.230822000` in `Directory.Packages.props`, I get a lot of build errors like these:

CS0433 The type 'Application' exists in both 'Microsoft.WinUI, Version=3.0.0.0, Culture=neutral, PublicKeyToken=de31ebe4ad15742b' and 'Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null'	UnoWizardApp1 (net7.0-windows10.0.19041.0)	C:\Repos\UnoWizardApp1\UnoWizardApp1\App.cs	3	24	Compiler	N/A

This is weird because afaik Uno.UI is for UWP, which is not a dependency in this solution - we are targeting WinUI.

- How can I fix these errors?

- Is there a reason NOT to update the Windows App SDK in a Uno sln?
  (I know the Uno API definition is based on older version of the Windows App SDK, but at least I'd like to update to benefit from fixes in the Windows App SDK)

Thanks for any help!

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. This got fixed in https://github.com/unoplatform/uno/pull/13521, and it will be backported soon. In the meantime, you can work around it by adding `<WindowsAppSDKWinUI>true</WindowsAppSDKWinUI>` in any `net7.0-windows` target.

**Reply by VincentH-Net** [reply]
Thanks! Adding that line to `<name>.Windows.csproj` and to `Directory.Build.props` (at location shown below) worked :-)
```xml
		<When Condition="$(TargetFramework.Contains('windows10'))">
			<PropertyGroup>
				...
				<WindowsAppSDKWinUI>true</WindowsAppSDKWinUI> <!-- Workaround for https://github.com/unoplatform/uno/discussions/13569 -->				
			</PropertyGroup>
		</When>
```

VS gets easily confused after making these edits - the ritual dance of close & reopen, clean, restore, rebuild helped there.
